hawtio-system/src/main/java/io/hawt/web/FileUploadFilter.java

```diff
@@ -0,0 +1,171 @@
+package io.hawt.web;
+
+import org.apache.commons.io.FileUtils;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+
+import static java.lang.StrictMath.toIntExact;
+
+public class FileUploadFilter {
+
+    private static final transient Logger LOG = LoggerFactory.getLogger(FileUploadFilter.class);
+    public static final String FILE_UPLOAD_PROPNAME = "hawtio.config.fileUpload";
+    private List<MagicNumberFileFilter> filters;
+
+    private void addFilters(MagicNumberFileFilter config) {
+        if (filters == null) {
+            this.filters = new ArrayList<>();
+        }
+
+        filters.add(config);
+    }
+
+    public FileUploadFilter() {
+
+        String config = System.getProperty(FILE_UPLOAD_PROPNAME);
+        if (config != null) {
```

Comment:
No default config?  Can any file be uploaded by default?

Reply:
With the current solution on detecting via magic numbers it would be a little tricky to blacklist some files out of the box as some file signatures might be different for the same file type, unless we add all possible scenario which could essentially clutter our code base, so I decided to delegate this to users themselves on what content type they really expect to accept via file upload so they can configure this that way.